### PR TITLE
Add .compile() and .compileAsync()

### PR DESCRIPTION
**Pull request by RobLoach**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "index.js"
   ],
   "devDependencies": {
-    "test-jstransformer": "^1.0.0"
+    "test-jstransformer": "jstransformers/test-jstransformer#async"
```
Comment (RobLoach) [inline package.json]: Currently needs the async fix for the tests.

File: index.js [hunk 2]
```diff
@@ -18,19 +18,28 @@ function getEnvironment (options) {
...
 exports.outputFormat = 'html';
 
-exports.render = function (str, options, locals) {
+exports.compile = function (str, options) {
   var env = getEnvironment(options)
-  return env.renderString(str, locals)
+  var tmpl = new Template(str, env, true)
+  return function (locals) {
+    return tmpl.render(locals)
+  }
```
Comment (TimothyGu) [inline index.js]: Can't you just `return tmpl.render`?

File: index.js [hunk 3]
```diff
@@ -18,19 +18,28 @@ function getEnvironment (options) {
...
-      if (err) {
-        return reject(err)
-      }
-      fulfill(res)
+exports.compileAsync = function (str, options) {
+  return new Promise(function (resolve, reject) {
+    var env = getEnvironment(options)
+    var tmpl = new Template(str, env, true)
+    resolve(function (locals) {
+      return new Promise(function (done, fail) {
```
Comment (TimothyGu) [inline index.js]: …it doesn't work this way.

```js
        compileAsync = (str, options) => Promise => (locals) => output
nunjacksCompileAsync = (str, options) => Promise => (locals) => Promise => output
```

Maybe it'll be better to just change compileAsync's signature to nunjacks'?

File: index.js (hunk 2 above)
Comment (RobLoach) [inline index.js]: It expected `this` to be set. Returning `tmpl.render` won't do that unfortunately.

Comment (TimothyGu) [inline index.js]: Okay.

Comment (RobLoach) [inline index.js]: We could do `return tmpl.render.bind(tmpl)`!

File: index.js (hunk 3 above)
Comment (RobLoach) [inline index.js]: > Maybe it'll be better to just change compileAsync's signature to nunjacks'?

How would we go about doing that?

Another acceptable one would be:

``` javascript
newCompileAsync = (str, options) => (locals) => Promise => output
```

**Pull request closed without merging**